File: benchmarks/java/integer_boolean/Java_Big_Integer/LiftExec.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;

public class LiftExec {



    private Lift _Lift = new Lift();



    private BInteger counter;

    public LiftExec() {
        counter = (BInteger) new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("3000"))).booleanValue()) {
            BInteger i = null;
            i = (BInteger) new BInteger("0");
            while((i.less(new BInteger("100"))).booleanValue()) {
                this._Lift.inc();
                i = (BInteger) i.plus(new BInteger("1"));
            }
            BInteger _i = null;
            _i = (BInteger) new BInteger("0");
            while((_i.less(new BInteger("100"))).booleanValue()) {
                this._Lift.dec();
                _i = (BInteger) _i.plus(new BInteger("1"));
            }
            counter = (BInteger) counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        LiftExec exec = new LiftExec();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: benchmarks/C++_Primitives/LiftExec.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"
#include "Lift.cpp"

#ifndef LiftExec_H
#define LiftExec_H

using namespace std;

class LiftExec {

    public:

    private:


        Lift _Lift;


        BInteger counter;

    public:

        LiftExec() {
            counter = (BInteger(0));
        }

        void simulate() {
            while((counter.less((BInteger(10000000)))).booleanValue()) {
                BInteger i;
                i = (BInteger(0));
                while((i.less((BInteger(100)))).booleanValue()) {
                    this->_Lift.inc();
                    i = i.plus((BInteger(1)));
                }
                BInteger _i;
                _i = (BInteger(0));
                while((_i.less((BInteger(100)))).booleanValue()) {
                    this->_Lift.dec();
                    _i = _i.plus((BInteger(1)));
                }
                counter = counter.plus((BInteger(1)));
            }
        }

        BInteger getCounter() {
            BInteger out;
            out = counter;
            return out;
        }

};

int main() {
    LiftExec exec;
    exec.simulate();
    return 0;
}

#endif